### PR TITLE
Update demo to use new Dash theme

### PR DESCRIPTION
**Pull request by thisac**
Updates the demo to use the new Dash theme:
* Refactors `app.py` so that it works similarly to the other updated demos.
* Adds a `custom.css` file updating the interface of the demo.
* Updates requirements to work correctly with added features.
* Update the tables slightly to make them easier to understand.

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -11,700 +11,304 @@
...
+    # one less to account for trainee
+    num_employees -= 1
+
+    shifts = [str(i + 1) for i in range(NUM_DAYS_IN_MONTH)]
+    df = utils.build_random_sched(num_employees, shifts, rand_seed)
+
+    init_availability_table = utils.display_availability(df, NOW.month, NOW.year)
+    return (
+        init_availability_table,
+        init_availability_table,
```
Comment (k8culver) [inline app.py:242]: I noticed that before this function was returning two different things here, what's the reasoning for having them be the same?

File: app.py [hunk 2]
```diff
@@ -11,700 +11,304 @@
...
+        (Output("cancel-button", "style"), {"display": "inline-block"}, {"display": "none"}),
+        (Output("run-button", "style"), {"display": "none"}, {"display": "inline-block"}),
+        # switch to schedule tab while running
+        (Output("schedule-tab", "disabled"), False, False),
+        (Output("tabs", "value"), "schedule-tab", "schedule-tab"),
+    ],
+    cancel=[Input("cancel-button", "n_clicks")],
+    prevent_initial_call=True,
 )
 def submitter(
```
Comment (k8culver) [inline app.py:270]: I wonder if we should come up with a consistent name for all these functions across the demos? Let me know your thoughts

File: app.py [hunk 3]
```diff
@@ -11,700 +11,304 @@
...
-            utils.display_schedule(sched, availability, month, year),
-            new_card_body,
-            "sched",
+            utils.display_schedule(sched, availability, NOW.month, NOW.year),
+            False,
         )
 
-    else:
-        return no_update, no_update, no_update
+    return no_update
```
Comment (k8culver) [inline app.py:311]: Oh that's cool, I didn't know you could just return one no_update for the whole thing

File: app_html.py [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+                options=[
+                    {"label": "Allow isolated days off", "value": 0},
+                    {"label": "Require exactly one manager on every shift", "value": 1},
+                ],
+                value=[1],
+                id="checklist-input",
+            ),
+            html.Div(
+                children=[
+                    html.Label("Random seed (optional)"),
```
Comment (k8culver) [inline app_html.py:139]: Might be good to indicate in the UI what this is used for

**Review by k8culver — COMMENTED** [review]
Looks really good! Nice code clean up decisions too!

Most of these comments are just small things/typos.

Couple other things I didn't know where to comment:
- In the JSS demo I added `padding-left: 6px !important;` to all the sliders and that makes them line up nicely with their titles, might be good to add that here too.
- Also just the funky height and width we chatted about yesterday

File: app_html.py [hunk 5]
```diff
@@ -0,0 +1,243 @@
+# Copyright 2024 D-Wave Systems Inc.
```
[the remark this answers is not in the document]
Comment (thisac) [reply app_html.py:1]: I agree. No strong preferences here. Maybe a _slight_ preference to having all app-related Python files named `app_....py` which also orders them nicely in file explorers, but happy either way.

File: app.py (hunk 1 above)
Comment (thisac) [reply app.py:242]: It displays the availability table in the schedule tab for when the results are loading (so the greyed out table while running). The tab is disabled in this same return and only enabled when the schedule tab is loading and after the results are displayed, so you should never see the availability table in the schedule tab except when greyed out during loading.

File: app.py (hunk 2 above)
Comment (thisac) [reply app.py:270]: Good idea. I think it's generally fine having slightly different names between demos if it makes functionality clearer, but if they are functionally the same we should try to be consistent with naming.

File: app_html.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (k8culver) [reply app_html.py:1]: Yeah I'd agree with that reasoning, I'm happy with that plan

File: app.py (hunk 1 above)
Comment (k8culver) [reply app.py:242]: Oh got it, that makes sense, thanks for clarifying!

File: app.py (hunk 3 above)
Comment (thisac) [reply app.py:311]: Oh, I didn't know that either (and I'm not sure this is actually ever called). I probably just forgot to change to `raise PreventUpdate` or `return no_update, no_update`. We could probably just remove this along with the `if ctx.triggered_id == "run-button"` altogether since the only input is the run-button.

File: app.py (hunk 2 above)
Comment (thisac) [reply app.py]: Changed to `run_optimization`, same as in MVRP.

File: app.py [hunk 6]
```diff
@@ -11,700 +11,306 @@
...
+        availability,
+        shifts,
+        *shifts_per_employee,
+        *employees_per_shift,
+        manager_required,
+        isolated_days_allowed,
+        consecutive_shifts + 1,
+    )
+
+    feasible_sampleset, errors = employee_scheduling.run_cqm(cqm)
```
Comment (k8culver) [inline app.py:300]: Might be for a different PR but should we do something with `errors` here?

File: assets/custom.css [hunk 7]
```diff
@@ -1,8 +1,367 @@
...
+
+.rc-slider-handle,
+.rc-slider-dot-active {
+    border-color: var(--blue-light);
+}
+
+.rc-slider-handle:active,
+.rc-slider-handle:hover,
+.rc-slider-handle-click-focused,
+.rc-slider-handle-click-focused:focus {
```
Comment (k8culver) [inline assets/custom.css:176]: This is just removing the blue hover effect when it's disabled 

```suggestion
.rc-slider:not(.rc-slider-disabled) .rc-slider-handle:active,
.rc-slider:not(.rc-slider-disabled) .rc-slider-handle:hover,
.rc-slider:not(.rc-slider-disabled) .rc-slider-handle-click-focused,
.rc-slider:not(.rc-slider-disabled) .rc-slider-handle-click-focused:focus {
```

File: assets/custom.css [hunk 8]
```diff
@@ -1,8 +1,367 @@
...
+  display: flex;
+}
+
+#left-column > div > div {
+  padding: 0 2rem;
+  width: 38rem;
+}
+
+#left-column > div:first-child {
+  width: 42rem;
```
Comment (k8culver) [inline assets/custom.css:144]: If this is changed to `36rem` and line 140 to `32rem` it makes everything fit a bit better on laptop screens. I personally don't think it makes the sidebar too small but if you feel like it does feel free to ignore this.

**Review by k8culver — COMMENTED** [review]
The scroll and height and width are way better, thanks for fixing that! Just a couple more small things, but overall looks great

File: app_html.py [hunk 9]
```diff
@@ -0,0 +1,241 @@
...
+            html.Div(
+                children=[
+                    html.Label("Random seed (optional)"),
+                    dcc.Input(id="seed-select", type="number", min=0),
+                ]
+            ),
+            html.Div(
+                id="button-group",
+                children=[
+                    html.Button(id="run-button", children="Solve", n_clicks=0),
```
[the remark this answers is not in the document]
Comment (thisac) [reply app_html.py:144]: I think it's fine if they differ as long as it makes sense in that demo and it's clear what the button does. I'm also fine with using the same text in all/most demos, again as long as it makes sense for the specific demo.

File: app.py (hunk 6 above)
Comment (thisac) [reply app.py:300]: Yeah, they were shown before, and it might make sense to show them somewhere still (perhaps in another hidden div). I'll leave that for the next PR though.

**Review by k8culver — COMMENTED** [review]
Couple more tiny things but I think after those it's good to go 🚀

**Review by k8culver — APPROVED** [review]
🚀
